### PR TITLE
Refactor release flow and use `changeset` to manage versioning

### PR DESCRIPTION
**Pull request by TuvalSimha**
- [x] Add `changeset` @TuvalSimha 
- [x] Add `renovate-bot` to the repo @kamilkisiela 
- [x] Add `NPM_TOKEN` secret @kamilkisiela 
- [x] Remove the old `0.0.0-PLACEHOLDER` and replace with `3.4.0` @TuvalSimha  
- [x] Set `publishConfig` if needed @TuvalSimha 
- [x] Remove old release scripts @TuvalSimha 
- [x] Test on PR (using `changeset` file) @TuvalSimha 
- [x] Make sure `fixedVersions` actually works as we expect  @TuvalSimha  
- [ ] Test stable release (from `master`)  @TuvalSimha

### CODE REVIEW

File: .changeset/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Changesets
```
Comment (dotansimha) [inline .changeset/README.md:1]: you can remove this one

File: .changeset/config.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "$schema": "https://unpkg.com/@changesets/config@2.3.0/schema.json",
+  "changelog": ["@changesets/changelog-github", { "repo": "kamilkisiela/graphql-inspector" }],
+  "commit": false,
+  "fixed": [["@graphql-inspector/*"]],
```
Comment (dotansimha) [inline .changeset/config.json:5]: have you tested this one?

File: .github/workflows/pr.yml [hunk 3]
```diff
@@ -12,4 +12,17 @@ jobs:
...
+      source: "Inspector"
+  release:
+    uses: the-guild-org/shared-config/.github/workflows/release-snapshot.yml@main
+    with:
+      npmTag: alpha
+      buildScript: build
+      nodeVersion: 18
+    secrets:
+      githubToken: ${{ secrets.GITHUB_TOKEN }}
+      npmToken: ${{ secrets.NPM_TOKEN }}
```
Comment (dotansimha) [inline .github/workflows/pr.yml:24]: This one is actually available?

File: .github/workflows/pr.yml [hunk 4]
```diff
@@ -12,4 +12,17 @@ jobs:
...
     with:
       domain: https://www.the-guild.dev/graphql/inspector/
-      source: 'Inspector'
+      source: "Inspector"
+  release:
+    uses: the-guild-org/shared-config/.github/workflows/release-snapshot.yml@main
+    with:
+      npmTag: alpha
+      buildScript: build
+      nodeVersion: 18
```
Comment (dotansimha) [inline .github/workflows/pr.yml:21]: This is the same version you are using in the other CI pipelines?

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  push:
+    branches:
+      - master
+
+jobs:
+  stable:
+    uses: the-guild-org/shared-config/.github/workflows/release-stable.yml@main
+    with:
+      releaseScript: release 
+      nodeVersion: 18
```
Comment (dotansimha) [inline .github/workflows/release.yml:12]: Same here, make sure it's the same one

File: package.json [hunk 6]
```diff
@@ -74,6 +76,10 @@
     "ts-jest": "27.0.5",
     "typescript": "4.2.4"
   },
+  "publishConfig": {
```
Comment (dotansimha) [inline package.json:79]: I think the root `package.json` doesn't need that

Comment (dotansimha) [reply package.json:79]: only internal packages might need? depends on how they are structured?

File: .changeset/config.json [hunk 7]
```diff
@@ -0,0 +1,14 @@
+{
+  "$schema": "https://unpkg.com/@changesets/config@2.3.0/schema.json",
+  "changelog": ["@changesets/changelog-github", { "repo": "kamilkisiela/graphql-inspector" }],
+  "commit": false,
+  "fixed": [],
```
Comment (dotansimha) [inline .changeset/config.json:5]: @TuvalSimha remove this key, also the `linked`

File: .github/workflows/pr.yml [hunk 8]
```diff
@@ -13,3 +13,16 @@ jobs:
...
+  release:
+    uses: the-guild-org/shared-config/.github/workflows/release-snapshot.yml@main
+    with:
+      npmTag: alpha
+      buildScript: build
+      nodeVersion: 16
+    secrets:
+      githubToken: ${{ secrets.GITHUB_TOKEN }}
+      npmToken: ${{ secrets.NPM_TOKEN }}
+  # dependencies:
```
Comment (dotansimha) [inline .github/workflows/pr.yml:25]: @TuvalSimha keep this commented out, but let's bring it back later, in a separate PR

**Review by dotansimha — APPROVED** [review]

**Review by kamilkisiela — APPROVED** [review]

**Pull request merged**